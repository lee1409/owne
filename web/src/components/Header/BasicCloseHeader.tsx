import { Box } from "@material-ui/core"
import HeaderContainer from "./HeaderContainer"
import { IconButton } from "../Button"

const BasicCloseHeader = () => (
  <HeaderContainer>
    <Box>
      <IconButton icon="Close" edge="start" />
    </Box>
  </HeaderContainer>
)

export default BasicCloseHeader
