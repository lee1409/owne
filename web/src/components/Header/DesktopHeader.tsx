import { HeaderLogo } from "../Typography"
import { Avatar, Box } from "@material-ui/core"
import SearchBar from "../SearchBar"
import HeaderNavList from "../HeaderNavList"
import HeaderContainer from "./HeaderContainer"

const DesktopHeader = () => (
  <HeaderContainer>
    <Box>
      <HeaderLogo />
    </Box>
    <Box ml={10} flexGrow={1}>
      <HeaderNavList />
    </Box>
    <Box mr={10}>
      <SearchBar />
    </Box>
    <Box>
      <Avatar />
    </Box>
  </HeaderContainer>
)

export default DesktopHeader
