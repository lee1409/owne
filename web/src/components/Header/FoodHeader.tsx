import { Box } from "@material-ui/core"
import { IconButton } from "../Button"
import HeaderContainer from "./HeaderContainer"

export const FoodMobileHeader = () => {
  return (
    <HeaderContainer>
      <Box flexGrow={1}>
        <IconButton icon="ArrowLeft" edge="start" />
      </Box>
      <Box>
        <IconButton icon="CircleEditOutline" edge="end" />
      </Box>
    </HeaderContainer>
  )
}
