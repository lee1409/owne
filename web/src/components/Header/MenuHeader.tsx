import { Box } from "@material-ui/core"
import { IconButton } from "../Button"
import { HeaderTitle } from "../Typography"
import HeaderContainer from "./HeaderContainer"

export const MenuMobileHeader = () => {
  return (
    <HeaderContainer>
      <Box flexGrow={1}>
        <HeaderTitle>Menu</HeaderTitle>
      </Box>
      <Box>
        <IconButton icon="Search" />
        <IconButton icon="AddCircle" edge="end" />
      </Box>
    </HeaderContainer>
  )
}
