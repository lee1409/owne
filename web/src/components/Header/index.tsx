import { AppBar, Toolbar, CssBaseline, Avatar } from "@material-ui/core"
import { styled } from "@material-ui/core/styles"
import { HeaderTitle, HeaderLogo } from "../Typography"
import { IconButton } from "../Button"
import SearchBar from "../SearchBar"
import HeaderNavList from "../HeaderNavList"
import { useMediaQueryUp } from "../../utils/hook"

const StyledToolBar = styled(Toolbar)(({ theme }) => ({
  margin: theme.spacing(5, 10),
}))

const StyledAppBar = styled(AppBar)(({ theme }) => ({
  backgroundColor: theme.palette.primary.main,
  color: theme.palette.secondary.main,
}))

const StyledHeaderNavList = styled(HeaderNavList)(({ theme }) => ({
  flexGrow: 1,
  padding: theme.spacing(0, 7),
  "& button": {
    marginLeft: theme.spacing(3),
  },
}))

const StyledSearchBar = styled(SearchBar)(({ theme }) => ({
  marginRight: theme.spacing(10),
}))

const StyledHeaderTitle = styled(HeaderTitle)(() => ({
  flexGrow: 1,
}))

const Header = () => {
  const matchedLg = useMediaQueryUp("lg")

  if (matchedLg) {
    return (
      <>
        <CssBaseline />
        <StyledAppBar elevation={0}>
          <StyledToolBar disableGutters>
            <HeaderLogo />
            <StyledHeaderNavList />
            <StyledSearchBar />
            <Avatar alt="Test User" />
          </StyledToolBar>
        </StyledAppBar>
      </>
    )
  }

  return (
    <>
      <CssBaseline />
      <StyledAppBar elevation={0}>
        <StyledToolBar disableGutters>
          <StyledHeaderTitle label="Menu" />
          <IconButton icon="search" />
          <IconButton icon="add" />
        </StyledToolBar>
      </StyledAppBar>
    </>
  )
}

export default Header
