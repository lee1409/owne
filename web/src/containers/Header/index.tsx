import { useLocation } from "react-router"
import DesktopHeader from "./DesktopHeader"
import { FoodMobileHeader } from "./FoodHeader"
import { MenuMobileHeader } from "./MenuHeader"
import ROUTE_NAME from "../../Router/routeName"
import { useMediaQueryUp } from "../../utils/hook"
import { getSegment } from "../../utils/urlHelper"

const Header = () => {
  const matchedDesktop = useMediaQueryUp("lg")
  const location = useLocation()

  // Desktop Header
  if (matchedDesktop) {
    return <DesktopHeader />
  }

  // Mobile headers
  switch (getSegment(location.pathname, 1)) {
    case ROUTE_NAME.FOOD:
      return <FoodMobileHeader />
    case ROUTE_NAME.MENU:
      return <MenuMobileHeader />
    default:
      return null
  }
}

export default Header
