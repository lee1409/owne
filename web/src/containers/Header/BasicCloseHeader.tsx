import { Box } from "@material-ui/core"
import HeaderContainer from "./HeaderContainer"
import { IconButton } from "../../components/Button"
import { FC } from "react"

type BasicCloseHeaderProps = {
  onClose: () => null
}

const BasicCloseHeader: FC<BasicCloseHeaderProps> = props => (
  <HeaderContainer>
    <Box>
      <IconButton icon="Close" edge="start" onClick={props.onClose} />
    </Box>
  </HeaderContainer>
)

export default BasicCloseHeader
