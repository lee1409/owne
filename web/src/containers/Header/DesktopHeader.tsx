import { HeaderLogo } from "../../components/Typography"
import { Avatar, Box } from "@material-ui/core"
import SearchBar from "../../components/SearchBar"
import HeaderNavList from "../../components/HeaderNavList"
import HeaderContainer from "./HeaderContainer"

const DesktopHeader = () => (
  <HeaderContainer>
    <Box>
      <HeaderLogo />
    </Box>
    <Box ml={10} flexGrow={1}>
      <HeaderNavList />
    </Box>
    <Box mr={10}>
      <SearchBar />
    </Box>
    <Box>
      <Avatar />
    </Box>
  </HeaderContainer>
)

export default DesktopHeader
