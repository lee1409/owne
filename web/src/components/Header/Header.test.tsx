import { render } from "@testing-library/react"
import Header from "."

test("Header", () => {
  const header = render(<Header />)
  expect(header).toMatchSnapshot("Header")
})
