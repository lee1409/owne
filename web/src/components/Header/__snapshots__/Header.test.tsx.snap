// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary WithStyles(ForwardRef(AppBar))-root-1 WithStyles(ForwardRef(AppBar))-root-2 MuiPaper-elevation0"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular WithStyles(ForwardRef(Toolbar))-root-3 WithStyles(ForwardRef(Toolbar))-root-4"
        >
          <p
            class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-7 HeaderTitle-root-5 HeaderTitle-root-6 MuiTypography-body1"
          >
            Menu
          </p>
          <button
            class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-8 WithStyles(ForwardRef(IconButton))-root-9"
            icon="Search"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                variant="Search"
                viewBox="0 0 24 24"
              >
                <span />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-8 WithStyles(ForwardRef(IconButton))-root-10 MuiIconButton-edgeEnd"
            icon="AddCircle"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                variant="AddCircle"
                viewBox="0 0 24 24"
              >
                <span />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionSticky MuiAppBar-colorPrimary WithStyles(ForwardRef(AppBar))-root-1 WithStyles(ForwardRef(AppBar))-root-2 MuiPaper-elevation0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular WithStyles(ForwardRef(Toolbar))-root-3 WithStyles(ForwardRef(Toolbar))-root-4"
      >
        <p
          class="MuiTypography-root WithStyles(ForwardRef(Typography))-root-7 HeaderTitle-root-5 HeaderTitle-root-6 MuiTypography-body1"
        >
          Menu
        </p>
        <button
          class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-8 WithStyles(ForwardRef(IconButton))-root-9"
          icon="Search"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              variant="Search"
              viewBox="0 0 24 24"
            >
              <span />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root WithStyles(ForwardRef(IconButton))-root-8 WithStyles(ForwardRef(IconButton))-root-10 MuiIconButton-edgeEnd"
          icon="AddCircle"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              variant="AddCircle"
              viewBox="0 0 24 24"
            >
              <span />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
